sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -400,12 +407,12 @@ def tune(
         trial_template = models.V1beta1TrialTemplate(
             primary_container_name=constants.DEFAULT_PRIMARY_CONTAINER_NAME,
             retain=retain_trials,
-            trial_parameters=trial_params,
+            trial_parameters=trial_params if callable(objective) else [],
             trial_spec=trial_spec,
         )
 
         # Add parameters to the Katib Experiment.
-        experiment.spec.parameters = experiment_params
+        experiment.spec.parameters = experiment_params if callable(objective) else []
```

Comment:
I think `parameters` field is also needed since `trial_paramaters` is still required.

WDTY👀 @akhilsaivenkata @andreyvelich

Reply:
You are right @Electronic-Waste , I have just reverted these two code changes and pushed it now.